### PR TITLE
Fix issue 56 - Check User before Delete

### LINKED ISSUE #56: Check for Exception not working as expected
When removing a Discorduser the Check for Not-Found-Exception isnt working as intended

```ts
  remove(id: number) {
    const updatedDiscord = this.discord.filter((user) => user.id !== id);

    //this checks if an Array is undefined, not if the Element to delete is present
    if (!updatedDiscord) { 
      throw new HttpException('User Not Found', HttpStatus.NOT_FOUND);
    }
    this.discord = [...updatedDiscord];
    return 'User deleted successfully!';
  }
```

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #56 

- [x] checks if user exists before deleting him

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Oh awesome 👍 This looks great, could we add a test for this? If you need help let me know as the tests are new and will probably need more functionality as the application grows 🤓

**Comment by Cahllagerfeld** [comment]
Would you suggest waiting until the tests from #54 are initialized merged, or should I go on creating this test?

**Comment by eddiejaoude** [comment]
Up to you, I think the e2e tests from here will not need the update step definitions from the other PR

File: test/features/discord.feature
Comment (Cahllagerfeld) [inline test/features/discord.feature:55]: Added Test for deleting non-existent user

**Review by eddiejaoude — APPROVED** [review]
Awesome 👍

File: test/features/discord.feature
Comment (eddiejaoude) [reply test/features/discord.feature:55]: Great! thank you 👍

**Pull request merged**
